### ISSUE TITLE
MacOS 上的 permission denied 问题，也可能和 cgo 有关

### ISSUE THREAD

**Issue by myzhan** [agiledragon/gomonkey#142]
环境
1. macOS Sonoma
2. AMD64
3. go 1.20.7

问题
之前都是正常使用，通过 cgo 引入 sqlite 依赖后，运行出现 permission denied 的问题。引入 cgo 的模块，跟 monkey patch 的模块没有任何关联。只是发现引入 cgo 后，monkey patch 的函数地址变得很大，40 多亿的一个整数，没有引入 cgo 之前是 4 千万。

解法
1. 通过 -ldflags=-extldflags=-Wl,-segprot,__TEXT,rwx,rx 参数运行 go test，可行。
2. 换用 https://github.com/bytedance/mockey 库，可行。通过对比代码，发现 mockey 在通过 mprotect 修改失败后，还会通过 mach_vm_protect 再尝试修改。去掉 mach_vm_protect 部分后，mockey 也不能 patch。

**Comment by agiledragon** [comment]
please review https://github.com/agiledragon/gomonkey/pull/143

**Comment by myzhan** [comment]
Tried on both intel machine and M1 machine, no luck.

**Comment by agiledragon** [comment]
Can you propose a PR to port mockey's mach_vm_protect feature to gomonkey?

**Comment by agiledragon** [comment]
https://github.com/agiledragon/gomonkey/pull/144   has been merged, please test it on both intel machine and M1 machine.

**Comment by myzhan** [comment]
Yes, I have tested it on both intel and M1 machine, macOS sonoma.

**Comment by myzhan** [comment]
Can you push a release tag?

**Comment by agiledragon** [comment]
> Yes, I have tested it on both intel and M1 machine, macOS sonoma.

Good job!

**Comment by agiledragon** [comment]
> Can you push a release tag?

v2.11.0 has been released!
